torch/utils/data/_utils/worker.py

```diff
@@ -327,3 +334,17 @@ def _worker_loop(dataset_kind, dataset, index_queue, data_queue, done_event,
     if done_event.is_set():
         data_queue.cancel_join_thread()
         data_queue.close()
+
+
+def _generate_seed(generator):
+    return torch.empty((), dtype=torch.int64, device="cpu").random_(generator=generator).item()
+
+
+def _non_default_cpu_generators():
+    out = [o for o in gc.get_objects()]
+    out = [o for o in out if gc.is_finalized(o)]
```

Comment:
This resolves the `E   ReferenceError: weakly-referenced object no longer exists` from https://github.com/pytorch/pytorch/issues/107531.

`gc.is_finalized()` was released in 3.9 though.